logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt

```diff
@@ -0,0 +1,80 @@
+package com.wire.kalium.logic.data.team
+
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.network.api.AssetId
+import com.wire.kalium.network.api.teams.TeamsApi
+import com.wire.kalium.network.utils.NetworkResponse
+import com.wire.kalium.persistence.dao.TeamDAO
+import com.wire.kalium.persistence.dao.TeamEntity
+import io.mockative.Mock
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.oneOf
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class TeamRepositoryTest {
+
+    @Mock
+    private val teamDAO = mock(classOf<TeamDAO>())
+
+    @Mock
+    private val teamMapper = mock(classOf<TeamMapper>())
+
+    @Mock
+    private val teamsApi = mock(classOf<TeamsApi>())
+
+    private lateinit var teamRepository: TeamRepository
+
+    @BeforeTest
+    fun setUp() {
+        teamRepository = TeamDataSource(
+            teamDAO = teamDAO,
+            teamMapper = teamMapper,
+            teamsApi = teamsApi
+        )
+    }
+
+    @Test
+    fun givenSelfUserExists_whenGettingTeamInfo_thenTeamInfoShouldBeReturned() = runTest {
```

Comment:
We could add another test and assert if `teamDAO` was called to insert the mapped team info.

Something like `givenTeamApiReturnsSuccessfully_whenGettingTeamInfo_thenTeamInfoIsPersisted`.

Also, another couple of tests to make sure that errors in the API or DAO are propagated correctly would be nice:

* `givenTeamApiFails_whenGettingTeamInfo_thenTheFailureIsPropagated`
* `givenTeamDAOFails_whenGettingTeamInfo_thenTheFailureIsPropagated`